### PR TITLE
[FLINK-15139][table sql / client]misc end to end test failed  cause loss jars in converting to jobgraph

### PR DESCRIPTION
**Pull request by leonardBang**
misc end to end test failed  cause loss jars in converting to jobgraph.

## What is the purpose of the change

*This pull request fix  loss jars when convert from streamgraph to jobgraph in `ExecutionContext`.*

## Brief change log

  - *update file org.apache.flink.table.client.gateway.local.ExecutionContext.java*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java [hunk 1]
```diff
@@ -310,6 +316,14 @@ private static Configuration createExecutionConfig(
 		Configuration executionConfig = activeCommandLine.applyCommandLineOptionsToConfiguration(
 				commandLine);
 
+		try {
+			final ProgramOptions programOptions = new ProgramOptions(commandLine);
+			final ExecutionConfigAccessor executionConfigAccessor = ExecutionConfigAccessor.fromProgramOptions(programOptions, dependencies);
+			ConfigUtils.encodeCollectionToConfig(executionConfig, PipelineOptions.JARS, executionConfigAccessor.getJars(), URL::toString);
+		} catch (CliArgsException e) {
+			throw new SqlExecutionException("Invalid deployment run options.", e);
+		}
```
Comment (wuchong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java]: You can simplify this into one line:

```java
		ConfigUtils.encodeCollectionToConfig(
			executionConfig,
			PipelineOptions.JARS,
			dependencies,
			URL::toString);
```

We don't need to parse `commandLine` again here.

Comment (kl0u) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java]: Actually both the `ProgramOptions` and the `executionConfig` are needed for the `flinkConfig` to be complete. This is the same as the `CliFrontend.getEffectiveConfiguration()` method.

So I think that the re-parsing is correct but the:

```ConfigUtils.encodeCollectionToConfig(executionConfig, PipelineOptions.JARS, executionConfigAccessor.getJars(), URL::toString);```

should become:

```executionConfigAccessor.applyToConfiguration(executionConfig);``` (in this order). 

And the `ExecutionConfigAccessor.applyToConfiguration()` should become public.

**Comment by kl0u** [comment]
Thanks for working on this @leonardBang !

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java (hunk 1 above)
Comment (leonardBang) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java]: @wuchong @kl0u thanks for your review, I'll update a new version soon.

**Comment by kl0u** [comment]
Hi @leonardBang . In the second commit you have unrelated changes in the `TableSourceUtil` and the `CatalogTableITCase`. Please remove them as they are not related to this PR.

**Comment by leonardBang** [comment]
@kl0u ok, the second commit fix a `DataTypes` issue which will hang this end to end test so I want fix this PR.
I‘ll open a new issue and fix the datatypes issue.

**Comment by leonardBang** [comment]
@kl0u Could you have a more look? and the other commit will create a new PR to [FLINK-15200](https://issues.apache.org/jira/browse/FLINK-15200)

**Comment by kl0u** [comment]
Thanks for the work @leonardBang ! Merged on master and release-1.10

**Pull request closed without merging**